### DISCUSSION TITLE
Future-Proofing Idea: "is_vital = (1, 0)" flag for systems involved in official quest lines.

### DISCUSSION THREAD

**Discussion by DoublePants** [endless-sky/endless-sky#10201] [Ideas]
I was thinking about the fact that there are a number of systems in the game that need to be in certain states for certain quests to work, and I was wondering if there's a flag in place for those systems to indicate this status. I looked briefly through the code but didn't see any such flag.

Essentially this wouldn't do anything in the current state of the game, but with regards to future-proofing it should make it a lot easier for the creation of plugins and possibly for in-game warnings about quest breakage. For example, let's say someone makes a plugin that involves a war between the Republic and a new faction. If there are systems that need to be under Republic control for, say, the Free Worlds quest chain to work, those would have something like vital = 1 in their system listing, so the person making the plugin would immediately know if the plugin is compatible with the main quest line, and could better craft it to increase compatibility.

Another thing this flag could do is allow plugins and various quest lines to alter the flag easily in the event that the player has locked themselves out of one or more quest lines. For example, let's say the Republic quest line is added and mutually exclusive with the Free Worlds one. If the player starts the Republic ques line, the event could set all vital = 1 tags to vital = 0 for any systems which are only vital due to the Free Worlds line.

Lastly, if a plugin is ever introduced allowing any kind of dynamic ownership changes of stars, it could refer to the vital marker to determine whether or not a star would be eligible to change hands.

**Comment by MasterOfGrey** [comment]
This is an excellent idea that I wholeheartedly support.
